src/v2/Components/NavBar/NavItem.tsx

```diff
@@ -219,28 +174,30 @@ export const NavItem: React.FC<NavItemProps> = ({
         color={color}
         underlineBehavior="none"
         px={1}
-        className={className}
         display={display}
-        onClick={handleClick}
+        className={className}
         aria-label={label}
+        tabIndex={tabIndex}
+        role={role}
+        onClick={handleClick}
       >
         {!!Menu && href && <UnfocusableAnchor to={href} />}
         <Text variant="text" lineHeight="solid" color={color}>
           <NavItemInner height={25}>
-            {isFunction(navItemLabel)
+            {isFunction(children)
               ? // NavItem children can be called as renderProps so that contents
                 // can operate on UI behaviors (such as changing the color of an
                 // icon on hover).
-                navItemLabel({ hover: isVisible })
-              : navItemLabel}
+                children({ hover: isVisible })
+              : children}
           </NavItemInner>
         </Text>
       </HitArea>
 
       {showMenu && (
-        <AnimatedPanel style={animatedStyle} menuAnchor={menuAnchor}>
+        <NavItemPanel menuAnchor={menuAnchor} relativeTo={containerRef}>
```

Comment:
If I do have to touch any of these in the future I'll probably just drop this whole thing and replace it but for now it's simple enough to pass a ref of the parent and then internally we just negative margin-left with the parent's offsetLeft.